packages/core/src/helpers/auth.ts

```diff
@@ -76,3 +80,19 @@ export const getConfiguredAliases = (context: AuthContext) => {
   const [primaryAlias, ...secondaryAliases] = aliases;
   return { primaryAlias, secondaryAliases };
 };
+
+/**
+ * Get the state of current actor. This is useful for checking which screen
+ * to render: e.g. `getActorState(state).matches('confirmSignUp.edit').
+ */
+export const getActorState = (state: AuthMachineState) => {
+  return state.context.actorRef?.getSnapshot();
+};
+
+/**
+ * Get the context of current actor. Useful for getting any nested context
+ * like remoteError.
+ */
+export const getActorContext = (state: AuthMachineState) => {
+  return getActorState(state)?.context;
+};
```

Comment:
Right, we need to chain `.actorRef?.getSnapshot()` because context like `remoteError` is nested under `state.context`. Top level machine only has these contexts: https://github.com/aws-amplify/amplify-ui/blob/322c001eb2412efb0af08aa6594f5b28965cce84/packages/core/src/types/authMachine.ts#L7-L13